pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -394,6 +398,15 @@ func collectCpuResourceRequestsFromDomainResources(vm *k6tv1.VirtualMachine) []o
 	cpuRequests := vm.Spec.Template.Spec.Domain.Resources.Requests.Cpu()
 
 	if cpuRequests == nil || cpuRequests.IsZero() {
+		// If no CPU requests and no Domain CPU are set, the VMI will default to 1 core
+		if vm.Spec.Template.Spec.Domain.CPU == nil {
```

Comment:
Fold into the conditional above?

Reply:
how so? we still want to skip the rest if only `cpuRequests == nil || cpuRequests.IsZero()`

Reply:
Apologies I misread the diff, thought the return was outside of the top conditional.

Reply:
@sradco is it worth the extra effort?